### ISSUE TITLE
[Bug]: GSheet plugin: User is denied access for read / update datasource and plugin objects 

### ISSUE THREAD

**Issue by sumitsum** [appsmithorg/appsmith#17503]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

- A user has shared an application where the action run on page load fails due to the following error: 
`Get_All_blankColor_Rows is not correctly configured. Please fix the following and then re-run: {Unable to find plugin 60910df653a4d211ac7dd678,Unable to find datasource 62fe750484757e0c055563b1}`
  - The reported objects are found to be present in the database. 
  - Based on the above points it can be inferred that one of the possibilities is that the user is denied read access to the concerned objects. 
  - In order to confirm the above idea I deleted the contents of `policies` key from a datasource object in my local and I was able to get the exact same error on action update then execute action. 
 - App: [https://app.appsmith.com/app/robby-s-thesys-workspace/blank-manage-633dc509dca6ec1f8c[…]heets-plugin/api/631b88ec238888080c597081?branch=Development](https://app.appsmith.com/app/robby-s-thesys-workspace/blank-manage-633dc509dca6ec1f8c6e4825/edit/saas/google-sheets-plugin/api/631b88ec238888080c597081?branch=Development)

### Steps To Reproduce

1. Open the link provided above and see the error on page load. 

### Version

cloud

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_5w8zw)

**Comment by Nikhil-Nandagopal** [comment]
@hiteshjoshi @trishaanand this caused a users existing app to suddenly stop working

**Comment by mohanarpit** [comment]
@sumitsum Thanks for raising the issue. Can you please help with the following information: 
1. Do we know if this API was working sometime back & suddenly started to fail? 
2. Did the user start seeing this failure after the application was connected to Git?
I'm asking for this information in order to triage the problem better.

**Comment by sumitsum** [comment]
> @sumitsum Thanks for raising the issue. Can you please help with the following information:
> 
> 1. Do we know if this API was working sometime back & suddenly started to fail?
> 2. Did the user start seeing this failure after the application was connected to Git?
>    I'm asking for this information in order to triage the problem better.

Hi Arpit, 
(1) The user has not explicitly mentioned so but the way he has phrased his problem it does seem like the queries were working before and stopped working later on. 
(2) The user has not mentioned connecting to Git at any point. However, we can confirm this directly with the user as well. The user is reachable via email. I have sent an email to the user with the questions and also added you to the recipient list.

**Comment by sumitsum** [comment]
> > @sumitsum Thanks for raising the issue. Can you please help with the following information:
> > 
> > 1. Do we know if this API was working sometime back & suddenly started to fail?
> > 2. Did the user start seeing this failure after the application was connected to Git?
> >    I'm asking for this information in order to triage the problem better.
> 
> Hi @mohanarpit , (1) The user has not explicitly mentioned so but the way he has phrased the problem it does seem like the queries were working before and stopped working later on. (2) The user has not mentioned connecting to Git at any point. However, we can confirm this directly with the user as well. The user is reachable via email. I have sent an email to the user with the questions and also added you to the recipient list.

User has responded with the following: 
```
the queries were working perfectly and then did suddenly stop working and i did connect to git but that was a while back and they were working while git was connected
```

**Comment by Nikhil-Nandagopal** [comment]
@hiteshjoshi @sidhantgoel do we have a timeline for this fix?

**Comment by sidhantgoel** [comment]
@Nikhil-Nandagopal I am working on debugging this issue, will share updates soon.

**Comment by sidhantgoel** [comment]
Seems like data is corrupted, we can push a migration to rectify the data, but root cause is still unknown. May be caused by git sync. I'll consult with other developers and update. Sorry for the delay.

**Comment by sumitsum** [comment]
@sidhantgoel can you please share what kind of data corruption, if possible ?

**Comment by sidhantgoel** [comment]
@sumitsum If you inspect `isInvalid` field in `unpublishedAction` field of action 'Get_All_Blank_Rows' in 'Development' branch, it is set to `false` while all the data is correct, and it shouldn't be invalid.

**Comment by Nikhil-Nandagopal** [comment]
@sidhantgoel @sumitsum whats the update here?

**Comment by sidhantgoel** [comment]
I think it happened due to race condition in import/export flow, I'm working on ImportExport code refactor, that should solve this issue as well, and for existing apps migrations will be pushed to correct the data. will update with ETA soon.

**Comment by infinitetrooper** [comment]
Closing this as the race condition issue was fixed in v1.9.18. Please reopen if this issue recurs.
